### PR TITLE
fix(metrics_extraction): Error related fields should not create on-demand metrics

### PR DESCRIPTION
**Pull request by armenzg**
Any time a query uses error-related fields (e.g. `issue`, `level`, etc) we should prevent creating an on-demand metric since this is a field that only error events support.

### CODE REVIEW

File: src/sentry/snuba/metrics/extraction.py [hunk 1]
```diff
@@ -779,7 +779,7 @@ def _is_standard_metrics_field(field: str) -> bool:
 
 
 def _is_error_field(token: str) -> bool:
-    return token.startswith("error.")
+    return token.startswith("error.") or token.startswith("level")
```
Comment (armenzg) [inline src/sentry/snuba/metrics/extraction.py:782]: Without this, `_get_query_supported_by(query)` would say that the query `level:foo` is an on-demand supported metric and `should_use_on_demand_metrics` would return `True`.
https://github.com/getsentry/sentry/blob/2b021cc5f45e3bbe6b46573abacf39a473eb8e5e/src/sentry/snuba/metrics/extraction.py#L589

File: tests/sentry/snuba/metrics/test_extraction.py [hunk 2]
```diff
@@ -166,24 +166,18 @@ class TestCreatesOndemandMetricSpec:
             ("percentile(transaction.duration, 1)", "transaction.duration>0"),
             ("count_if(transaction.duration,equals,0)", "transaction.duration:>0"),
             ("count_if(transaction.duration,notEquals,0)", "transaction.duration:>0"),
-            (
-                "count()",
-                "project:a-1 route.action:CloseBatch level:info",  # custom tags not supported by standard metrics
```
Comment (armenzg) [inline tests/sentry/snuba/metrics/test_extraction.py:171]: I'm only removing `level:info` since it not anymore a supported on-demand query.

**Review by k-fish — APPROVED** [review]
Looks good, but let's do a few while we're here and have a constant we can refer to if people ask why on-demand isn't extracting in edge cases where the values overlap (overlap occurs if someone adds a tag that has the same name as a default event field, not that would normally happen but it _could_)

**Review by obostjancic — APPROVED** [review]

File: src/sentry/snuba/metrics/extraction.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (armenzg) [reply src/sentry/snuba/metrics/extraction.py]: I handled it:

```diff
diff --git a/src/sentry/snuba/metrics/extraction.py b/src/sentry/snuba/metrics/extraction.py
index 56f5026e1e..746343650e 100644
--- a/src/sentry/snuba/metrics/extraction.py
+++ b/src/sentry/snuba/metrics/extraction.py
@@ -95,6 +95,10 @@ QUERY_HASH_KEY = "query_hash"
 # TODO: Streamline with dynamic sampling.
 RuleCondition = Union["LogicalRuleCondition", "ComparingRuleCondition", "NotRuleCondition"]

+# There are some search tokens that are exclusive to searching errors, thus, we need
+# to treat the query as not on-demand.
+ERROR_RELATED_TOKENS = ["level", "assignee", "issue", "culprit"]
+
 # Maps from Discover's field names to event protocol paths. See Relay's
 # ``Getter`` implementation for ``Event`` for supported fields. All fields need to be prefixed
 # with "event.".
@@ -779,7 +783,7 @@ def _is_standard_metrics_field(field: str) -> bool:


 def _is_error_field(token: str) -> bool:
-    return token.startswith("error.") or token.startswith("level")
+    return token.startswith("error.") or token in ERROR_RELATED_TOKENS
```

**Pull request merged**